# **ESTIMATION AND CONFIDENCE INTERVALS**

In [38]:
import numpy as np
import scipy.stats as stats
import math

In [39]:
data = np.array([
    1.13, 1.55, 1.43, 0.92, 1.25,
    1.36, 1.32, 0.85, 1.07, 1.48,
    1.20, 1.33, 1.18, 1.22, 1.29
])

print("Print-head durability data:")
print(data)

Print-head durability data:
[1.13 1.55 1.43 0.92 1.25 1.36 1.32 0.85 1.07 1.48 1.2  1.33 1.18 1.22
 1.29]


In [ ]:
#mean
n = len(data)

sample_mean = np.mean(data)

print("Sample size:", n)
print("Sample mean:", sample_mean)

# 99% Confidence Interval Using Sample Standard

In [40]:
sample_std = np.std(data, ddof=1)

print("Sample standard deviation:", sample_std)

ddof=1

Sample standard deviation: 0.19316412956959936


In [42]:
n = len(data)

confidence_level = 0.99
alpha = 1 - confidence_level
df = n - 1

t_critical = stats.t.ppf(1 - alpha / 2, df)

print("Degrees of freedom:", df)
print("t-critical value:", t_critical)

Degrees of freedom: 14
t-critical value: 2.976842734370834


In [43]:
#Calculate Standard Error
standard_error = sample_std / math.sqrt(n)

print("Standard error:", standard_error)

#Calculate Margin of Error
margin_of_error = t_critical * standard_error

print("Margin of error:", margin_of_error)

Standard error: 0.04987476379384733
Margin of error: 0.14846932822817596


In [45]:
sample_mean = np.mean(data)

#Calculate 99% Confidence Interval
lower_limit = sample_mean - margin_of_error
upper_limit = sample_mean + margin_of_error

print("99% Confidence Interval:")
print("Lower limit:", lower_limit)
print("Upper limit:", upper_limit)

99% Confidence Interval:
Lower limit: 1.0901973384384906
Upper limit: 1.3871359948948425


Why is the t-distribution used?

The t-distribution is used because the population standard deviation is unknown and the sample size is small (n = 15). Instead of using the population standard deviation, the sample standard deviation is used to estimate the population variability. The t-distribution accounts for this additional uncertainty and uses n - 1 = 14 degrees of freedom.

99% Confidence Interval Using Known Population Standard Deviation

In [46]:
population_std = 0.2

print("Known population standard deviation:", population_std)

z_critical = stats.norm.ppf(1 - alpha / 2)

print("z-critical value:", z_critical)

standard_error_z = population_std / math.sqrt(n)

print("Standard error:", standard_error_z)

margin_of_error_z = z_critical * standard_error_z

print("Margin of error:", margin_of_error_z)

lower_limit_z = sample_mean - margin_of_error_z
upper_limit_z = sample_mean + margin_of_error_z

print("99% Confidence Interval using known population standard deviation:")
print("Lower limit:", lower_limit_z)
print("Upper limit:", upper_limit_z)

Known population standard deviation: 0.2
z-critical value: 2.5758293035489004
Standard error: 0.051639777949432225
Margin of error: 0.13301525327090588
99% Confidence Interval using known population standard deviation:
Lower limit: 1.1056514133957607
Upper limit: 1.3716819199375725


Using sample standard deviation and t-distribution:
99% CI = (1.0902, 1.3871)

Using known population standard deviation and z-distribution:
99% CI = (1.1057, 1.3717)

# **Conclusion**
The sample of 15 print-heads has an average durability of approximately 1.2387 million characters. When the population standard deviation is unknown, the 99% confidence interval using the t-distribution is approximately 1.0902 to 1.3871 million characters. When the population standard deviation is known to be 0.2 million characters, the 99% confidence interval using the z-distribution is approximately 1.1057 to 1.3717 million characters. The t-distribution is appropriate for the first case because the sample size is small and the population standard deviation is unknown.

# **HYPOTHESIS TESTING**

In [57]:
import numpy as np
from scipy import stats

sample_mean = 3050
sample_size = 25

units_mean = 600
units_std = 25

alpha = 0.05

theoretical_mean = 1000 + (5 * units_mean)

print("Theoretical mean weekly cost:", theoretical_mean)


cost_std = 5 * units_std

print("Standard deviation of weekly cost:", cost_std)

#Calculate Standard Error
standard_error = cost_std / np.sqrt(sample_size)

print("Standard error:", standard_error)

#Calculate Test Statistic
z_statistic = (
    sample_mean - theoretical_mean
) / standard_error

print("Test statistic (z):", z_statistic)

#Calculate Critical Value
critical_value = stats.norm.ppf(1 - alpha)

print("Critical value:", critical_value)

#Calculate p-value
p_value = 1 - stats.norm.cdf(z_statistic)

print("p-value:", p_value)

#Decision Rule
if z_statistic > critical_value:
    print("Reject the null hypothesis.")
else:
    print("Fail to reject the null hypothesis.")

#p-value Decision
if p_value < alpha:
    print("Reject the null hypothesis.")
else:
    print("Fail to reject the null hypothesis.")


Theoretical mean weekly cost: 4000
Standard deviation of weekly cost: 125
Standard error: 25.0
Test statistic (z): -38.0
Critical value: 1.6448536269514722
p-value: 1.0
Fail to reject the null hypothesis.
Fail to reject the null hypothesis.


# **Conclusion**
At the 5% significance level, the test statistic is -38, while the critical value for the right-tailed test is approximately 1.6449. Since the test statistic is less than the critical value, we fail to reject the null hypothesis. The p-value is approximately 1.0, which is also greater than the significance level of 0.05.

Therefore, there is no statistical evidence to support the restaurant owners' claim that the weekly operating cost is higher than the theoretical model suggests. In fact, the provided sample mean of Rs. 3,050 is lower than the theoretical mean cost of Rs. 4,000.